[[ NO MERGE ]] Livecodeali/tests run on buildbot

livecodeali opened the pull request.

livecodeali committed:
[[ WIP ]] Test libs tweaks for windows test runner

livecodeali committed:
[[ ExtensionUtils ]] Tweak for packaging test extensions with no docs

livecodeali commented:
@livecode-vulcan review ok d5d104b

livecode-vulcan commented:
💙 review by @livecodeali ok d5d104b6ebd7824ab142cf8fe0997e244267278c

livecode-vulcan commented:
😞 test failure d5d104b6ebd7824ab142cf8fe0997e244267278c

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/467)
* try-community-armv7-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/459)
* try-community-arm64-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/459)
* try-community-x86-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/459)
* try-community-x86_64-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/459)
* try-community-js-emscripten-sdk1.35: [pending](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1981)
* try-community-universal-ios-iphoneos12.1: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/209)
* try-community-universal-ios-iphonesimulator12.1: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/208)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2012)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2003)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2001)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2035)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1972)

livecodeali commented:
@livecode-vulcan retry d5d104b

livecodeali commented:
@livecode-vulcan review ok f3916ab

livecode-vulcan commented:
💙 review by @livecodeali ok f3916ab1300c8030fece7bba061dc7a1d5d66b11

livecode-vulcan commented:
😞 test failure f3916ab1300c8030fece7bba061dc7a1d5d66b11

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/468)
* try-community-armv7-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/460)
* try-community-arm64-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/460)
* try-community-x86-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/460)
* try-community-x86_64-android-sdk26_ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/460)
* try-community-js-emscripten-sdk1.35: [pending](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1982)
* try-community-universal-ios-iphoneos12.1: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/210)
* try-community-universal-ios-iphonesimulator12.1: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/209)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2013)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2004)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2002)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2036)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1973)

livecodeali committed:
[[ WindowsCI ]] Add windows support to test makefiles

This patch tweaks the various path calculations in test makefiles
(and the common makefile) to support testing windows executables

livecodeali committed:
[[ WindowsCI ]] Add editbin replacement script

This patch adds a python script which twiddles the appropriate bit
in a windows executable to make it a console application. Console 
applications do not open a separate console window when invoked on 
the command line - the command prompt only returns when the executable 
process finishes. This is the necessary behavior for the standalone
engine when running tests.

livecodeali committed:
[[ WindowsCI ]] Fix tests for windows

This patch fixes, removes or skips a number of tests that fail or
are no longer correct on windows.

livecodeali committed:
[[ WindowsCI ]] Allow use of chocolatey-installed winflexbison

This patch allows the use of flex and bison which have been 
installed via the chocolatey package winflexbison as the method
of installing via cygwin does not work on Travis' windows 
containers. Use of win_flex and win_bison can be controlled with
an environment variable `USE_WINFLEXBISON`

livecodeali committed:
[[ WindowsCI ]] Allow skipping of lcidlc

On Travis' windows container, the generation of lcidlc's support files
fails and causes a compilation error. This patch adds a mechanism for
skipping lcidlc builds.

livecodeali committed:
[[ WindowsCI ]] Allow make.cmd to be run from base folder

This patch removes the need to run make.cmd from within the 
generated build folder.

livecodeali committed:
[[ WindowsCI ]] Don't pause configure script on warning

This patch ensures that if we are running on Travis, the
configure script does not pause for warnings as this requires
user input and we want the build to continue even if there
are warnings.

livecodeali committed:
[[ WindowsCI ]] Add travis configuration file

This patch adds a travis configuration file for windows. Currently
this build uses the chocolatey winflexbison package, skips
building lcidlc and only runs LiveCode script based tests, and
is restricted to extension and engine tests for now.

livecodeali committed:
[[ WIP ]] Add build infrastructure to run tests
